server/src/db.mjs

```diff
@@ -17,57 +40,41 @@ export const addOrUpdateUser = (user) =>
     user,
   );
 
-export const addOrUpdateStock = (stock) =>
+export const addStockToWatchlist = (sub, ticker) =>
   db.one(
-    `
-    INSERT INTO stocks(ticker, updated_at, company_name, market_cap, PE_ratio,
-      week52_high, week52_low, YTD_change, volume, latest_price, change_percent)
-    VALUES($<symbol>, NOW(), $<companyName>, $<marketCap>, $<peRatio>,
-      $<week52High>, $<week52Low>, $<ytdChange>, $<volume>, $<latestPrice>,
-      $<changePercent>)
-    ON CONFLICT (ticker) DO UPDATE
-      SET updated_at = NOW(), market_cap = $<marketCap>, PE_ratio = $<peRatio>,
-      week52_high = $<week52High>, week52_low = $<week52Low>,
-      YTD_change = $<ytdChange>, volume = $<volume>,
-      latest_price = $<latestPrice>, change_percent = $<changePercent>
-    RETURNING *
-    `,
-    stock,
+    `INSERT INTO watchlist(user_id, ticker)
+      VALUES((SELECT id FROM users WHERE sub=$<sub>), $<ticker>)
+      ON CONFLICT (ticker) DO NOTHING
+      RETURNING *`,
+    { sub, ticker },
   );
 
-export const getWatchlist = (sub) =>
-  db.any(
-    "SELECT watchlist.*, stocks.* FROM watchlist LEFT JOIN stocks on stock_id=stocks.id LEFT JOIN users on user_id=users.id WHERE sub=$<sub>",
-    { sub },
+export const addUserPortfolio = (sub, portfolio) =>
+  db.one(
+    `INSERT INTO user_portfolio(user_id, portfolio_name, time_period, portfolio_values)
+    VALUES((SELECT id FROM users WHERE sub=$<sub>), $<portfolioName>, $<timePeriod>, ARRAY [$<initialAmount>])
+      RETURNING *`,
+    { sub, ...portfolio },
   );
 
-export const addStockToWatchlist = (sub, ticker) =>
+export const addPortfolioStocks = (portfolioID, ticker, allocation) =>
   db.one(
-    `INSERT INTO watchlist(user_id, stock_id)
-      VALUES((SELECT id FROM users WHERE sub=$<sub>), (SELECT id FROM stocks WHERE ticker=$<ticker>))
+    `INSERT INTO portfolio_stock(portfolio_id, ticker, allocation)
+    VALUES($<portfolioID>, $<ticker>, $<allocation>)
       RETURNING *`,
-    { sub, ticker },
+    { portfolioID, ticker, allocation },
   );
 
 export const deleteStockFromWatchlist = (sub, ticker) =>
   db.none(
-    "DELETE FROM watchlist WHERE user_id = (SELECT id FROM users WHERE sub=$<sub>) AND stock_id = (SELECT id FROM stocks WHERE ticker=$<ticker>)",
+    "DELETE FROM watchlist WHERE user_id = (SELECT id FROM users WHERE sub=$<sub>) AND ticker=$<ticker>",
     { sub, ticker },
   );
 
-export const getPortfolios = (sub) =>
-  db.any(
-    "SELECT user_portfolio.* FROM user_portfolio LEFT JOIN users on user_id=users.id WHERE sub=$<sub>",
-    { sub },
-  );
-
-export const addUserPortfolio = (sub) =>
-  db.one(
-    `INSERT INTO user_portfolio(user_id)
-      VALUES((SELECT id FROM users WHERE sub=$<sub>))
-      RETURNING *`,
-    { sub },
-  );
+export const deletePortfolio = (portfolioID) =>
+  db.none("DELETE FROM user_portfolio WHERE portfolio_id = $<portfolioID>", {
```

Comment:
When you delete a portfolio shouldn't you also delete the portfolio's stock? Or is postgres doing that for you with a `DELETE CASCADE`?

Reply:
Yes! I have `ON UPDATE CASCADE ON DELETE CASCADE` in postgres.